src/main/kotlin/minesweeper/domain/model/Board.kt

```diff
@@ -0,0 +1,24 @@
+package minesweeper.domain.model
+
+class Board private constructor(
+    private val cells: List<List<Cell>>
+) : List<List<Cell>> by cells {
+
+    init {
+        val flatCells = cells.flatten()
+
+        require(!flatCells.all { cell -> cell.isMine() }) {
+            "보드의 모든 셀이 지뢰여서는 안된다."
+        }
+    }
+
+    companion object {
+        fun create(width: Width, height: Height, mineCount: MineCount): Board {
+            val cells = List(width.toInt() * height.toInt()) { index -> Cell.create(index < mineCount.toInt()) }
+                .shuffled()
+                .chunked(width.toInt())
```

Comment:
아래와 같이 mineCells 를 먼저 만들고 noneCells를 따로 만든 뒤, 하나로 합쳐준 뒤, 동일한 로직을 태우면 조금 더 코드를 읽기가 쉬워질 것 같아요. 😃

```suggestion
            val mineCells = List(mineCount.toInt()) { Cell.MINE }
            val noneCells = List((width.toInt() * height.toInt()) - mineCount.toInt()) { Cell.NONE }

            val cells = (mineCells + noneCells)
                .shuffled()
                .chunked(width.toInt())
```

Reply:
그렇네요 제가보기에도 한눈에 정리가 된게 잘보이네요!
감사합니다!!